### PR TITLE
dash-mpd-cli: init at 0.2.23

### PR DESCRIPTION
**Pull request by al3xtjames**
This PR adds [dash-mpd-cli](https://github.com/emarsden/dash-mpd-cli) to nixpkgs.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [x] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/da/dash-mpd-cli/package.nix [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+      file, as used for on-demand replay of TV content and video streaming
+      services.
+    '';
+    homepage = "https://emarsden.github.io/dash-mpd-cli/";
+    downloadPage = "https://github.com/emarsden/dash-mpd-cli/releases";
+    changelog = "https://github.com/emarsden/dash-mpd-cli/blob/main/CHANGELOG.md";
+    license = licenses.mit;
+    maintainers = with maintainers; [ al3xtjames ];
+    mainProgram = "dash-mpd-cli";
+    platforms = platforms.all;
```
Comment (Aleksanaa) [inline pkgs/by-name/da/dash-mpd-cli/package.nix:46]:
```suggestion
  meta = {
    description = "Download media content from a DASH-MPEG or DASH-WebM MPD manifest";
    longDescription = ''
      A commandline application for downloading media content from a DASH MPD
      file, as used for on-demand replay of TV content and video streaming
      services.
    '';
    homepage = "https://emarsden.github.io/dash-mpd-cli/";
    downloadPage = "https://github.com/emarsden/dash-mpd-cli/releases";
    changelog = "https://github.com/emarsden/dash-mpd-cli/blob/main/CHANGELOG.md";
    license = lib.licenses.mit;
    maintainers = with lib.maintainers; [ al3xtjames ];
    mainProgram = "dash-mpd-cli";
```

(remove `platforms`, as `rustPlatform.buildRustPackage` carries available platforms)

**Review by Aleksanaa — APPROVED** [review]

File: pkgs/by-name/da/dash-mpd-cli/package.nix (hunk 1 above)
Comment (al3xtjames) [reply pkgs/by-name/da/dash-mpd-cli/package.nix]: Thanks! I updated the commit to remove  `meta.platforms`.

Comment (Aleksanaa) [reply pkgs/by-name/da/dash-mpd-cli/package.nix]: (remove `with lib`)

**Review by Aleksanaa — APPROVED** [review]

**Comment by Aleksanaa** [comment]
@ofborg build dash-mpd-cli

**Review by Aleksanaa — APPROVED** [review]

File: pkgs/by-name/da/dash-mpd-cli/package.nix [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+  src = fetchFromGitHub {
+    owner = "emarsden";
+    repo = "dash-mpd-cli";
+    rev = "v${version}";
+    hash = "sha256-gRtt7iocGmnFpdTEMv/U4izeR/NtdYYXX3eFXW5LGYs=";
+  };
+
+  postPatch = ''
+    ln -s ${./Cargo.lock} Cargo.lock
```
Comment (dotlambda) [inline pkgs/by-name/da/dash-mpd-cli/package.nix:21]: Please ask upstream to provide a lock file.

**Pull request merged**